src/lib.rs

```diff
@@ -140,7 +140,9 @@ fn rmain(config: &mut Config) -> Result<()> {
     if !no_message_format {
         cargo.arg("--message-format").arg("json-render-diagnostics");
     }
-    for arg in args {
+
+    let args = args.collect::<Vec<_>>();
```

Comment:
Why do we need to do this now?

Reply:
I need to use `args` later when the subcommand is `DownloadToolchain` to extract the version so I don't want it to be consumed here. I also did not want to search for the version here, since it's only related to the download toolchain subcommand. So, for clarity I think it's better for each subcommand to consume the args in a way the command expects it.

Of course, I wished we used clap from the beginning :)

Reply:
We can't use clap because we pass the "rest" of the args to cargo directly (not sure if that's possible with clap... it should be now I think about it), let's leave it at this for now.